_posts/2017-12-28-Secure-your-architecture-part2.md

```diff
@@ -0,0 +1,211 @@
+tags: [Microservices, Security, Encryption, Spring, Cloud, Config]
+category: Microservices
+comments: true
+---
+
+Since the rise of the digital era, most enterprises keep their data in a digital format.
+But if their sensitive data lacks security, it can cause the data to be unreliable, unstable and unavailable to their business.
+We have to be prepared if an attacker breaches in to our network and tries to hack our sensitive data.
+Whether it is in-motion or at-rest, encrypting our data and using the proper protection mechanism, will make it worthless for the hacker to use. 
+
+# Overview
+* [Cryptographic Algorithms](#cryptographic-algorithms)
+* [Key Mechanics](#key-mechanics)
+* [Cloud-hosted Key management service](#cloud-hosted-key-management-service)
+* [Spring Cloud Config Server](#spring-cloud-config-server)
+
+# Cryptographic Algorithms
+When implementing our application, every programming language will provide us with a set of known libraries for cryptographic algorithms. 
+A big flaw is implementing an algorithm by yourself, the known algorithms have been reviewed, patched and been known for their excellent security. 
+These are the most used types that you can use for encryption at rest: 
+
+## Symmetric Encryption
+The key used in encrypting data at rest is used for both encrypting and decrypting the data.
+This key becomes very vulnerable if anyone gets a hold on it. 
+* Well known: [Advanced Encryption Standard encryption](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard){:target="_blank"}
+
+
+## Asymmetric Encryption
+In asymmetric encryption, a pair of keys are used. One public that is exposed and encrypts your data and one private key that is only known by the owner that decrypts your data.
+* Wel known:  [Rivest–Shamir–Adleman encryption](https://en.wikipedia.org/wiki/RSA_(cryptosystem)){:target="_blank"}
+
+# Key Mechanics 
+Encryption keys are another aspect of encryption, handling the keys become just as sensitive as the data itself. 
+That's why we need mechanisms on how keys are stored and shared so attackers can't get a hold on them.
+
+## Key rotation
+Encryption key rotation will provide protection especially when the certificate expires, is corrupted or the key management admin is no longer part of the company. 
+Lets say, you got a good eye at detecting patterns and detect that the same key is being used for encrypting data.
+To avoid this, you rotate your keys, and every time the same data field is encrypted it will result in another encrypted message.
+
+## JSON Web Key (Set)
+We discussed in the previous post about retrieving a JWK(S) to verify our JSON Web Token in our microservice. 
+A JWK is a JSON object that represents a cryptographic key that consists of information to verify a JWT. 
+If you like to dive into Signing JSON documents your can check out this [post](https://ordina-jworks.github.io//security/2016/03/12/Digitally-signing-your-JSON-documents.html#jwk){:target="_blank"}
```

Comment:
signing (without capital letter)

Would also rephrase it a bit: 
... documents **you** can checkout the blog post on \[Digitally signing your JSON documents\](https://ordina-jworks.github.io//security/2016/03/12/Digitally-signing-your-JSON-documents.html#jwk){:target="_blank"}. 
(also don't forget the period at the end of the sentence)